x-pack/plugins/security_solution/server/lib/detection_engine/signals/filter_events_with_list.test.ts

```diff
@@ -106,6 +125,72 @@ describe('filterEventsAgainstList', () => {
         'ci-badguys.txt'
       );
       expect(res.hits.hits.length).toEqual(2);
+
+      // @ts-ignore
```

Comment:
Is it possible for us to use an `as cast` or better yet an `is cast` using the io-ts is on this here or do we have to keep the `ts-ignore`. We really should be more strict about allowing ts-ignore. Also there is `ts-expect-error` which it supposed to replace the ts-ignore:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments

it will be better in that builds will fail once it is no longer an error condition, but I prefer we try to not use ts-ignore as much as possible.

Reply:
the ts-ignore says that `.ip` is not available on `item._source.source` 

```
Property 'ip' does not exist on type 'string | number | boolean | object | string[] | number[] | boolean[] | object[]'.
```

comes from here https://github.com/elastic/kibana/blob/a7a2b7cb4c58e9d9228580e907dcbd1c88781006/x-pack/plugins/security_solution/server/lib/detection_engine/signals/types.ts#L31-L49

I'm not sure there is a way to change that without making a more explicit structure with every ECS property.